Connect the front and back end for Login and Registration

muhammedyusuf-sermet opened the pull request:
This pull request completes the login and registration workflow. Right now, the session tokens and page routing works fine. The testing is mediocre right now and will possibly increase coverage for the next sprints. The challenges with tests took an unreasonable amount of time and it's best to check this functionality in and not postpone it until we have perfect testing. The main functionality is tested so it's not too bad.

hekemp commented on frontend/src/renderer/components/LoginHeader.tsx:
```diff
@@ -1,20 +1,168 @@
+		var context = this;
+		var request = require("request");
+		var options = { method: 'POST',
+			url: 'http://3.18.65.138:3000/login',
```
This is planned to be a task for early in next sprint

muhammedyusuf-sermet commented on frontend/src/renderer/components/LoginHeader.tsx:
```diff
@@ -1,20 +1,168 @@
+	status: number,
+	message: string, 
+	token: string
+}
```
Good suggestion, changed the name

muhammedyusuf-sermet commented on frontend/src/renderer/components/LoginHeader.tsx:
```diff
@@ -1,20 +1,168 @@
+		request(options, function (error: string, response: string, body: requestBodyInterface) {
+			if (error) {
+				context.openSnackbar("There has been a server error. Please try again later.");
+				throw new Error(error);
```
Agreed, removed the throw. The wording 'request' not really a user-friendly word i believe, it's more technical. I modified the message the to emphasize the error in login.

muhammedyusuf-sermet commented on frontend/src/renderer/components/Platform.tsx:
```diff
@@ -0,0 +1,49 @@
+			event.preventDefault();
+			cookieManager.RemoveCookie("session_token");
+			this.handleRedirectToHome();
+		}
```
It's not actually between Login Header and Platform, it's between Home and Platform. We basically have two components in this application; everything else are their child.

Home: When user is not logged in
Platform: When user is logged in

I'll keep this as is at the moment and we can talk about our structure in the next sprint and make tasks accordingly.

muhammedyusuf-sermet commented:
Thanks for the review David. I addressed all the issues above and updated the code where necessary.

> Feels like it works, is there something in particular that I should be looking at?

Not necessarily.

> The code structure looks good however I don't see any use of redux. Are we abandoning that? Should we be using reacts "Provider" "Consumer" style, or keep the slightly awkward state transitions?

We didn't use Redux yet and I'll keep this consistent with the rest of the app. Let's talk about this in the scrum meting for the next sprint and decide how we want to go from here. Good point.

Feel free to go ahead and approve the changes
